pkg/skaffold/deploy/kubectl/kubectl.go

```diff
@@ -168,7 +175,8 @@ func (k *Deployer) trackNamespaces(namespaces []string) {
 
 // Deploy templates the provided manifests with a simple `find and replace` and
 // runs `kubectl apply` on those manifests
-func (k *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Artifact, manifests manifest.ManifestList) error {
+func (k *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Artifact, manifestsByConfig manifest.ManifestListByConfig) error {
+	manifests := manifestsByConfig[k.configName]
```

Comment:
should we use `k.ConfigName()` instead of `k.configName`? It might be better in some cases as these functions could potentially have some logic to handle special cases such as empty config names or something like that